homeassistant/components/intent/timers.py

```
@@ -232,7 +248,7 @@ def start_timer(
             name=f"Timer {timer_id}",
         )
 
-        for handler in self.handlers:
+        for handler in self.handlers[timer.device_id]:
```

All functions in `TimerManager` should raise if action is for an unsupported device ID.

---

That check shouldn't happen by the caller of these methods.